[Bug] S3 P2 Pipeline "Checkstyle" fails after gradle check

### Description

The pipeline [Checkstyle](https://github.com/asirbescu-fhbg/fh-burgenland-bswe-3b-2at/blob/main/.github/workflows/checkstyle.yml) configured for running the checkstyle tool over the code (see #5) fails at the job "Run ./gradlew check" (workflow run 6).

Error message:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'examination'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve org.springframework.boot:spring-boot-gradle-plugin:3.2.2.
     Required by:
         project : > org.springframework.boot:org.springframework.boot.gradle.plugin:3.2.2
      > No matching variant of org.springframework.boot:spring-boot-gradle-plugin:3.2.2 was found. The consumer was configured to find a library for use during runtime, compatible with Java 11, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '8.5' but:
          - Variant 'apiElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.2.2 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.5')
          - Variant 'javadocElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.2.2 declares a component for use during runtime, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 11)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.5')
          - Variant 'mavenOptionalApiElements' capability org.springframework.boot:spring-boot-gradle-plugin-maven-optional:3.2.2 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.5')
          - Variant 'mavenOptionalRuntimeElements' capability org.springframework.boot:spring-boot-gradle-plugin-maven-optional:3.2.2 declares a library for use during runtime, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component, compatible with Java 17 and the consumer needed a component, compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.5')
          - Variant 'runtimeElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.2.2 declares a library for use during runtime, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component, compatible with Java 17 and the consumer needed a component, compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.5')
          - Variant 'sourcesElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.2.2 declares a component for use during runtime, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 11)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.5')

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org./

BUILD FAILED in 7s
Error: Process completed with exit code 1.
```

### Severity

S3 - Minor Bug

### Priority

P2 - Medium

### To Reproduce

_No response_

### Expected behavior

_No response_

### Screenshots

![grafik](https://github.com/asirbescu-fhbg/fh-burgenland-bswe-3b-2at/assets/114183627/e5cb940e-48fb-4586-9181-80e13679c839)


### Additional Information

_No response_

--- a/.github/workflows/checkstyle.yml
+++ b/.github/workflows/checkstyle.yml
@@ -10,8 +10,13 @@ on:
     
 jobs:
   checkstyle:
-    name: Run checkstyle
+    name: Run checkstyle via gradle check
     runs-on: ubuntu-latest
     steps:
         - uses: actions/checkout@v4
+        - uses: actions/setup-java@v4
+          with:
+            distribution: oracle
+            java-version: 21
+        - name: Run gradle check command
         - run: ./gradlew check

--- a/.github/workflows/checkstyle.yml
+++ b/.github/workflows/checkstyle.yml
@@ -19,4 +19,4 @@ jobs:
             distribution: oracle
             java-version: 21
         - name: Run gradle check command
-        - run: ./gradlew check
+          run: ./gradlew check

